dev/src/transaction.ts

```diff
@@ -290,7 +288,7 @@ export class Transaction implements firestore.Transaction {
     documentRef: firestore.DocumentReference<AppModelType, DbModelType>,
     data: firestore.WithFieldValue<AppModelType>
   ): Transaction {
-    if (this._readOnly) {
+    if (!this._writeBatch) {
```

Comment:
@MarkDuckworth I have changed the logic a little, the `_readOnly` field no longer exists, with the following motivation
- `_writeBatch` is now initialised only when it is read-write to avoid initialising a write batch for readonly transactions (same for `_backoff`)
- then the `_readOnly` flag would be superfluous as the presence of `_writeBatch` is now the authoritative flag for whether it is readonly or read/write (so I removed `_readOnly`)
- and `_readOnlyReadTime` field is optional, just because a read time is not set doesn't mean it isn't readonly

Reply:
Thanks for the detailed reply. Sounds good then.